### PR TITLE
#65: Creating wallets.

### PR DESCRIPTION
**Pull request by paulodamaso**
For #65: Creating wallets.
- Added `createsRightWallet` test which asserts that the right wallet was created
- Added fake constructor with wallet parameters to `Wallet.Fake` so that `createsRightWallet` test compile
- Added method `WalletsIn.create` with parameters which will create `Wallet`
- Added puzzles for filling the added methods and fixing tests.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#78](https://github.com/zold-io/java-api/issues/78) is now in scope, role is `REV`

**Comment by paulodamaso** [comment]
@llorllale Would you please review and merge this one? This task is preventing me from receiving new tasks, and I've made this PR more than two months ago.

File: src/main/java/io/zold/api/Wallet.java [hunk 1]
```diff
@@ -122,6 +130,17 @@ public Fake(final long id, final Transaction... transactions) {
             this(id, new IterableOf<>(transactions));
         }
 
+        /**
+         * Complete Ctor.
```
Comment (llorllale) [inline src/main/java/io/zold/api/Wallet.java]: @paulodamaso please remove "Complete" from this javadoc

File: src/main/java/io/zold/api/Wallet.java [hunk 2]
```diff
@@ -122,6 +130,17 @@ public Fake(final long id, final Transaction... transactions) {
...
+         * @param id The wallet id.
+         * @param pubkey The wallet public key.
+         * @param network The network the walet belongs to.
+         * @checkstyle UnusedFormalParameter (2 lines)
+         */
+        public Fake(final long id, final String pubkey, final String network) {
```
Comment (llorllale) [inline src/main/java/io/zold/api/Wallet.java]: @paulodamaso I'm not sure what you intend `pubkey` to be. Take a look at [this](https://github.com/zold-io/java-api#decisions-and-alternatives).

Comment (paulodamaso) [reply src/main/java/io/zold/api/Wallet.java]: @llorllale As defined in [zold whitepaper](https://papers.zold.io/wp.pdf) page 9, wallets are composed of:
> 1. Network name, [a-z]{4,16} ;
> 2. Protocol version, [0-9]+ ;
> 3. Wallet ID, a 64-bit unsigned integer in hexadecimal format;
> 4. Public RSA key of the wallet owner, in Base64.
So, `pubkey` refers to the public RSA of the wallet owner.

**Comment by paulodamaso** [comment]
@llorllale Please take a look

File: src/main/java/io/zold/api/Wallet.java (hunk 2 above)
Comment (llorllale) [reply src/main/java/io/zold/api/Wallet.java]: @paulodamaso well, yes. My bad - I was not specific enough.

My concern was with the level of abstraction. We're using the `java.security` package, so this RSA public key could perhaps be better expressed as a parameter with type [RSAPublicKey](https://docs.oracle.com/javase/8/docs/api/java/security/interfaces/RSAPublicKey.html).

**Comment by llorllale** [comment]
@rultor merge

**Pull request merged**